c++/itertools/itertools.hpp

```diff
@@ -148,23 +169,14 @@ namespace itertools {
 
     /********************* Product Iterator ********************/
 
-    // Sentinel_t, used to denote the end of a product range
-    template <typename It>
-    struct sentinel_t {
-      It it;
-    };
-    template <typename It>
-    sentinel_t<It> make_sentinel(It &&it) {
-      return {it};
-    }
+    template <typename sentinel_tuple_t, typename... It>
```

Comment:
Also, why this dissymetry between the sentinels (passed as tuple) 
and the iterators, passed as pack. 
?

Reply:
Agreed on the naming. The asymmetry is because you cannot use two parameter packs!
Also for the sentinel we don't actually need the parameter pack in the implementation.
We only need it for the `its` when we `dereference`
```c++
return std::make_tuple(*std::get<Is>(its)...);
```